cirq-core/cirq/transformers/routing/mapping_manager.py

```diff
@@ -14,154 +14,196 @@
 
         Raises:
-            ValueError: whenever lq1 and lq2 are no adjacent on the device.
+            ValueError: whenever lq1 and lq2 are not adjacent on the device.
         """
         if self.dist_on_device(lq1, lq2) > 1:
             raise ValueError(
                 f"q1: {lq1} and q2: {lq2} are not adjacent on the device. Cannot swap them."
             )
 
-        pq1, pq2 = self._map[lq1], self._map[lq2]
-        self._map[lq1], self._map[lq2] = self._map[lq2], self._map[lq1]
-
-        self._inverse_map[pq1], self._inverse_map[pq2] = (
-            self._inverse_map[pq2],
-            self._inverse_map[pq1],
-        )
+        pq1, pq2 = self.logical_to_physical[lq1], self.logical_to_physical[lq2]
+        self._logical_to_physical[[lq1, lq2]] = self._logical_to_physical[[lq2, lq1]]
+        self._physical_to_logical[[pq1, pq2]] = self._physical_to_logical[[pq2, pq1]]
 
     def mapped_op(self, op: 'cirq.Operation') -> 'cirq.Operation':
-        """Transforms the given operation with the qubits in self._map.
+        """Transforms the given logical operation to act on corresponding physical qubits.
 
         Args:
-            op: an operation on logical qubits.
+            op: logical operation acting on logical qubits.
 
         Returns:
-            The same operation on corresponding physical qubits."""
-        return op.transform_qubits(self._map)
+            The same operation acting on corresponding physical qubits.
+        """
+        logical_ints = [self._logical_qid_to_int[q] for q in op.qubits]
+        physical_ints = self.logical_to_physical[logical_ints]
+        qubit_map: Dict['cirq.Qid', 'cirq.Qid'] = {
+            q: self._int_to_physical_qid[physical_ints[i]] for i, q in enumerate(op.qubits)
+        }
+        return op.transform_qubits(qubit_map)
 
-    def shortest_path(self, lq1: 'cirq.Qid', lq2: 'cirq.Qid') -> Sequence['cirq.Qid']:
-        """Find the shortest path between two logical qubits on the device given their mapping.
+    def shortest_path(self, lq1: int, lq2: int) -> Sequence[int]:
+        """Find the shortest path between two logical qubits on the device, given their mapping.
 
         Args:
-            lq1: the first logical qubit.
-            lq2: the second logical qubit.
+            lq1: integer corresponding to the first logical qubit.
+            lq2: integer corresponding to the second logical qubit.
 
         Returns:
-            A sequence of logical qubits on the shortest path from lq1 to lq2.
+            A sequence of logical qubit integers on the shortest path from `lq1` to `lq2`.
         """
-        return [
-            self._inverse_map[pq]
-            for pq in nx.reconstruct_path(self._map[lq1], self._map[lq2], self._predecessors)
+        return self.physical_to_logical[
+            nx.reconstruct_path(*self.logical_to_physical[[lq1, lq2]], self._predecessors)
         ]
-
-    def _value_equality_values_(self):
```

Comment:
What motivates the removal of these methods? It isn't addressed in the PR description.

Reply:
Updated the PR description with an explanation of why these methods are removed.